### PR TITLE
chore(deps): upgrade test deps

### PR DESCRIPTION
**Pull request by vvo**

### CODE REVIEW

File: test/next/tsconfig.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "Node",
+    "moduleResolution": "Bundler",
```
Comment (correttojs) [inline test/next/tsconfig.json:13]: I didn't know about this option 👀

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 20.x]
+        node-version: [18.x, 20.x]
```
Comment (correttojs) [inline .github/workflows/unit-tests.yml:22]: we still have 16 available in vercel, do we want to drop these tests? are they failing?

**Review by correttojs — APPROVED** [review]

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (vvo) [reply .github/workflows/unit-tests.yml:22]: Yep Next.js dropped Node 16 support in 14.x

File: test/next/tsconfig.json (hunk 1 above)
Comment (vvo) [reply test/next/tsconfig.json:13]: This is the default of Next.js tsconfig.json nowadays. BTW "Node" means module resolution is Node 10x lol..

**Pull request merged**